wire-ios-sync-engine/Source/Use cases/EnableAnalyticsUseCase.swift

```diff
@@ -33,31 +34,42 @@ public protocol EnableAnalyticsUseCaseProtocol {
 
 /// Concrete implementation of the EnableAnalyticsUseCaseProtocol.
 /// This struct is responsible for enabling analytics sharing for a specific user profile.
-public struct EnableAnalyticsUseCase: EnableAnalyticsUseCaseProtocol {
+struct EnableAnalyticsUseCase: EnableAnalyticsUseCaseProtocol {
 
     // MARK: - Properties
 
-    private let analyticsManager: AnalyticsManagerProtocol
+    private let analyticsSessionConfiguration: AnalyticsSessionConfiguration
+
+    private let analyticsManagerBuilder: (_ appKey: String, _ host: URL) -> any AnalyticsManagerProtocol
+
+    private let sessionManager: AnalyticsManagerProviding
 
     private let analyticsUserProfile: AnalyticsUserProfile
 
-    private let userSession: ZMUserSession
+    private let analyticsSessionProvider: EnableAnalyticsUseCaseUserSession
 
     // MARK: - Initialization
 
     /// Initializes a new instance of EnableAnalyticsUseCase.
     ///
     /// - Parameters:
-    ///   - analyticsManager: The analytics manager to use for enabling tracking.
+    ///   - analyticsManagerBuilder: A closure that provides an `AnalyticsManagerProtocol` implementation.
+    ///   - sessionManager: The session manager that conforms to `AnalyticsManagerProviding` for managing analytics sessions.
+    ///   - analyticsSessionConfiguration: The configuration for the analytics session.
     ///   - analyticsUserProfile: The user profile for which to enable analytics sharing.
-    public init(
-        analyticsManager: AnalyticsManagerProtocol,
+    ///   - analyticsSessionProvider: An instance conforming to `EnableAnalyticsUseCaseUserSession` that manages the user's analytics session state.
+    init(
```

Comment:
Since the `init` doesn't have to be `public`, it can be omitted:
Just declare the properties like this:
```
/// The configuration for the analytics session.
private(set) var analyticsSessionConfiguration: AnalyticsSessionConfiguration

/// A closure that provides an `AnalyticsManagerProtocol` implementation.
private(set) var analyticsManagerBuilder: (_ appKey: String, _ host: URL) -> any AnalyticsManagerProtocol

/// The session manager that conforms to `AnalyticsManagerProviding` for managing analytics sessions.
private(set) var sessionManager: AnalyticsManagerProviding

/// The user profile for which to enable analytics sharing.
private(set) var analyticsUserProfile: AnalyticsUserProfile

/// An instance conforming to `EnableAnalyticsUseCaseUserSession` that manages the user's analytics session state.
private(set) var analyticsSessionProvider: EnableAnalyticsUseCaseUserSession
```

Reply:
Done!